model_compression_toolkit/core/pytorch/mixed_precision/configurable_activation_quantizer.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2023 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from typing import Dict, List, Any
+
+from model_compression_toolkit.constants import FOUND_TORCH
+from model_compression_toolkit.core.common.mixed_precision.configurable_quant_id import ConfigurableQuantizerIdentifier
+from model_compression_toolkit.core.common.quantization.candidate_node_quantization_config import \
+    CandidateNodeQuantizationConfig
+from model_compression_toolkit.logger import Logger
+from model_compression_toolkit.target_platform_capabilities.target_platform import QuantizationMethod
+from mct_quantizers import QuantizationTarget
+from mct_quantizers import mark_quantizer
+
+
+if FOUND_TORCH:
+    import torch
+    import torch.nn as nn
+    from mct_quantizers.pytorch.quantizers import BasePyTorchInferableQuantizer
+
+    @mark_quantizer(quantization_target=QuantizationTarget.Activation,
+                    quantization_method=[QuantizationMethod.POWER_OF_TWO, QuantizationMethod.SYMMETRIC,
+                                         QuantizationMethod.UNIFORM, QuantizationMethod.LUT_POT_QUANTIZER],
```

Comment:
It looks pretty much the same as TensorFlow. why do need 2 implementations? Is there any unique code for Pytorch/Tensorflow?  at least consider using "common" because there is a lot in common.

Reply:
I agree that there is a lot of logic that looks similar in those classes, but it is mandatory to create a separate quantizer for each framework.
I'll try to move some of the shared code that is completely framework-independent to a common location, but if it would complicate the code I prefer to have some duplications in those to separate classes.